Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/review.py

```diff
@@ -232,4 +395,108 @@ def main(cls, args, repository, **kwargs):
             ], diff=pull_request.diff(comments=True),
         )
 
-        return 0
+        if args.dry_run:
+            if response:
+                sys.stderr.write('User modified the pull-request, but is running as a dry-run, no edits were published.\n')
+                return 1
+            return 0
+
+        returncode = 0
+        did_approve = None
+        reviewers_to_add = set()
+        user, _ = pull_request.generator.repository.credentials(required=False)
+
+        for key in ('Approved by', 'Blocked by'):
+            value = (response.get('metadata') or {}).pop(key, None)
+            if value is None:
+                continue
+            original = pull_request.approvers if key == 'Approved by' else pull_request.blockers
+            is_me, added, removed, rc = cls.user_delta(pull_request, original, value)
+            returncode += rc
+            if is_me:
+                did_approve = key == 'Approved by'
+            reviewers_to_add.update(added + removed)
+        if 'metadata' in response and not response['metadata']:
+            del response['metadata']
+
+        if pull_request.opened and did_approve is None and user and pull_request.author != user:
+            did_approve = dict(Yes=True, Block=False).get(Terminal.choose(
+                'Would you like approve this change?',
+                default='No', options=('Yes', 'No', 'Block'),
```

Comment:
Yes / No / Block does not feel very intuitive to me. In response to the question "Would you like to approve this change?", "No" sounds like r-, but it's actually what GitHub calls "Comment".

Could we use the same idioms as GitHub? Like: "How would you like to review this change?" with choices "Comment only", "Approve", "Request changes"